lib/fluent/plugin/in_tail.rb

```diff
@@ -909,19 +1004,40 @@ def opened?
 
         private
 
+        def sleep_for_log_ingestion(start_reading)
+          # sleep to stop reading files when we reach the read bytes per second limit, to throttle the log ingestion
+          time_spent_reading = Fluent::Clock.now - start_reading
+          @log.debug("time_spent_reading: #{time_spent_reading} #{ @watcher.path}")
+          if time_spent_reading < 1
+            needed_sleeping_time = 1 - time_spent_reading
+            @log.debug("sleep: #{needed_sleeping_time}")
+            sleep(needed_sleeping_time)
```

Comment:
Why it's needed to sleep? I think just abort reading the file and wait next timer event is more elegant.

Reply:
Although this feature requires `enable_watch_timer`:

https://github.com/fluent/fluentd/blob/af2ea9bb3af594bf0a1fbb18eaf701c26ebcf291/lib/fluent/plugin/in_tail.rb#L192-L194

I think it's not needed if we use `sleep` instead of waiting next timer.

Reply:
Fixed in https://github.com/fluent/fluentd/pull/3185/commits/d3a3023fc8e3d6b06fc37c7f432fc91b5694471f.